include/zephyr/kernel.h

```diff
@@ -6153,6 +6153,186 @@ void k_sys_runtime_stats_enable(void);
+ * @param sig the signal which should be removed from @a set.
+ *
+ * @retval 0 on success.
+ * @return -EINVAL if @a sig is invalid.
+ */
+int k_sig_delset(struct k_sig_set *set, int sig);
+
+/**
+ * @brief Check if signal @a sig is part of signal set @a set.
+ *
+ * @param set the set to check.
+ * @param sig the signal which should be checked.
+ *
+ * @retval 0 if @a sig is not part of @a set.
+ * @retval 1 if @a sig is part of @a set.
+ * @return -EINVAL if @a sig is invalid.
+ */
+int k_sig_ismember(const struct k_sig_set *set, int sig);
+
+/**
+ * @brief Initialize a signal set with the set of all signals.
+ *
+ * @param set signal set to initialize.
+ *
+ * @retval 0 on success.
+ * @return -EINVAL if @a sig is invalid.
+ */
+int k_sig_fillset(struct k_sig_set *set);
+
+/**
+ * @brief Initialize a signal set with the set of all signals.
+ *
+ * @param set signal set to initialize.
+ *
+ * @retval 0 on success.
+ * @return -EINVAL if @a sig is invalid.
+ */
+int k_sig_emptyset(struct k_sig_set *set);
+
+/**
+ * @brief Block signals in the accompanying @ref k_sig_set
+ *
+ * @see @ref k_sig_mask
+ */
+#define K_SIG_BLOCK 0
+
+/**
+ * @brief Set signals in the accompanying @ref k_sig_set
+ *
+ * @see @ref k_sig_mask
+ */
+#define K_SIG_SETMASK 1
+
+/**
+ * @brief Unblock signals in the accompanying @ref k_sig_set
+ *
+ * @see @ref k_sig_mask
+ */
+#define K_SIG_UNBLOCK 2
+
+/** @brief Minimum realtime signal number */
+#define K_SIG_RTMIN 32
+
+/** @brief Number of supported realtime signals */
+#define K_SIG_NUM_RT MAX(CONFIG_SIGNAL_SET_SIZE - K_SIG_RTMIN, 0)
+
+/**
+ * @brief Manipulate the mask of the calling thread.
+ *
+ * @param how how to manipulate the set
+ * @param set the set to manipulate
+ * @param[out] oset the resulting set
+ * @retval 0 on success
+ * @retval -EINVAL for invalid values of @a how
+ */
+__syscall int k_sig_mask(int how, const struct k_sig_set *set, struct k_sig_set *oset);
+
+/**
+ * @brief Get set of pending signals for the calling thread.
+ *
+ * @param[out] set the set of pending signals for the calling thread.
+ * @retval 0 on success
+ * @retval -EINVAL for invalid values of @a how
+ */
+__syscall int k_sig_pending(struct k_sig_set *set);
+
+/**
+ * @brief Queue a signal for process ID @a pid.
+ *
+ * Queue the signal @a signo for process ID @a pid with value @a value.
+ *
+ * This function is safe to call in ISR context.
+ *
+ * @param pid the process ID for which to queue the signal.
+ * @param signo the signal number to queue.
+ * @param value the value associated with the signal.
+ *
+ * @retval 0 on success.
+ * @retval -EAGAIN if there were insufficient resources within the specified @a timeout.
+ * @retval -EINVAL if @a signo is invalid.
+ * @return -EPERM if the calling thread lacks permissions to signal the process with id @a pid.
+ * @return -ESRCH if a process with id @a pid does not exist.
+ */
+__syscall int k_sig_queue(k_pid_t pid, int signo, union k_sig_val value);
```

Comment:
`k_sigqueue()`?

Reply:
Since Jukka also thought `k_sig_` was maybe preferable, I've left it that way (can revisit if needed).